### ISSUE TITLE
SA1619 false alarm on partial classes

### ISSUE THREAD

**Issue by sungam3r** [DotNetAnalyzers/StyleCopAnalyzers#2903]
SA1619 should consider documentation on partial classes.

File1.cs:

```c#
/// <summary>blablabla. </summary>
/// <typeparam name="TPaymentRequest"> aaa. </typeparam>
public abstract partial class BasePostRequestProcessor<TPaymentRequest>
{
}
```

File2.cs:

```c#
/// <summary>blablabla. </summary>
public abstract partial class BasePostRequestProcessor<TPaymentRequest> // SA1619	The documentation for type parameter 'TPaymentRequest' is missing
{
}
```

File2.cs after change:

```c#
 /// <summary>blablabla. </summary>
 /// <typeparam name="TPaymentRequest"> aaa. </typeparam> // CS1710 XML comment has a duplicate typeparam tag for 'TPaymentRequest'
public abstract partial class BasePostRequestProcessor<TPaymentRequest>
{
}
```

**Comment by sharwell** [comment]
File2.cs should not have a `<summary>` element. Only one part of the partial type should have this documentation element, and the analyzers assume this is the same part that will document type parameters. Updating File2.cs to use `<content>` instead of `<summary>` should resolve this problem. See the documentation page of [SA1619](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1619.md) for additional information.

**Comment by sharwell** [comment]
@sungam3r Thanks for reporting this. I filed #2908 as a new analyzer idea to help users avoid this situation.

**Comment by sungam3r** [comment]
Thanks! Indeed, I had to carefully examine the documentation for the rule.
